src/Testcontainers/Containers/Modules/Databases/CosmosDbTestcontainer.cs

```diff
@@ -0,0 +1,110 @@
+namespace DotNet.Testcontainers.Containers
+{
+  using System.Collections.Generic;
+  using System.Net.Http;
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Configurations;
+  using Microsoft.Extensions.Logging;
+  using System.Text.Json;
+  using System;
+  using System.Threading;
+  using System.Text;
+
+  public sealed class CosmosDbTestcontainer : TestcontainerDatabase
+  {
+    private string CosmosUrl;
+
+    private HttpClient HttpClient;
+
+    internal CosmosDbTestcontainer(ITestcontainersConfiguration configuration, ILogger logger)
+        : base(configuration, logger)
+    {
+        CosmosUrl = $"https://{this.Username}.documents.azure.com/dbs";
+    }
+
+    public override string ConnectionString
+        => $"AccountEndpoint=https://{this.Username}:{this.Port}.documents.azure.com:443/;AccountKey={this.Password}";
```

Comment:
`this.Username` --> `this.Hostname` (it is not `localhost` for every Docker environment).